src/org/zaproxy/zap/extension/openapi/OpenApiSpider.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.openapi;
+
+import net.htmlparser.jericho.Source;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.zap.extension.openapi.converter.Converter;
+import org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter;
+import org.zaproxy.zap.extension.openapi.network.Requestor;
+import org.zaproxy.zap.spider.parser.SpiderParser;
+
+public class OpenApiSpider extends SpiderParser {
+
+    private static final Logger log = Logger.getLogger(OpenApiSpider.class);
+    private Requestor requestor;
+    private static boolean isEnabled = false;
+
+    public OpenApiSpider() {
+        requestor = new Requestor(HttpSender.SPIDER_INITIATOR);
+        requestor.addListener(new HistoryPersister());
+    }
+
+    @Override
+    public boolean parseResource(HttpMessage message, Source source, int depth) {
+        if (!isEnabled) return false;
+
+        try {
+            Converter converter = new SwaggerConverter(message.getResponseBody().toString());
+            requestor.run(converter.getRequestModels());
+        } catch (Exception e) {
+            log.debug(e.getMessage(), e);
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean canParseResource(HttpMessage message, String path, boolean wasAlreadyConsumed) {
+        try {
+            String contentType = message.getResponseHeader().getHeader(HttpHeader.CONTENT_TYPE);
+            if (contentType.toLowerCase().startsWith("text") && 
+                    message.getResponseBody().toString().toLowerCase().startsWith("swagger:")) {
+                return true;
+            } else if (contentType.toLowerCase().startsWith("application")) {
+                return true;
+            }
+        } catch (Exception e) {
+            return false;
+        }
+        log.info("Cant parse " + message.getRequestHeader().getURI());
+        return false;
+    }
+
+    public static void enable() {
```

Comment:
These methods (`enable` and `disable`) could be removed (if using the 2.6.0 features).